lib/multiplexer/multiplexer_test.go

```diff
@@ -298,6 +298,50 @@ func TestMux(t *testing.T) {
 		require.NotNil(t, err)
 	})
 
+	// makes sure the connection get port set to 0
+	// when PROXY protocol is unspecified
+	t.Run("required PROXY line", func(t *testing.T) {
+		t.Parallel()
+		listener, err := net.Listen("tcp", "127.0.0.1:0")
+		require.NoError(t, err)
+
+		mux, err := New(Config{
+			Listener:              listener,
+			PROXYProtocolMode:     PROXYProtocolUnspecified,
+			IgnoreSelfConnections: true,
+		})
+		require.NoError(t, err)
+		go mux.Serve()
+		defer mux.Close()
+
+		backend1 := &httptest.Server{
+			Listener: mux.TLS(),
+			Config: &http.Server{
+				Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					fmt.Fprintf(w, r.RemoteAddr)
+				}),
+			},
+		}
+		backend1.StartTLS()
+		defer backend1.Close()
+
+		parsedURL, err := url.Parse(backend1.URL)
+		require.NoError(t, err)
+
+		conn, err := net.Dial("tcp", parsedURL.Host)
+		require.NoError(t, err)
+		defer conn.Close()
+
+		// upgrade connection to TLS
+		tlsConn := tls.Client(conn, clientConfig(backend1))
+		defer tlsConn.Close()
+
+		// make sure the TLS call failed
+		res, err := utils.RoundtripWithConn(tlsConn)
+		require.Nil(t, err)
+		require.Equal(t, conn.LocalAddr().String(), res)
```

Comment:
I'm not sure I understand what this test is doing... you upgrade to a TLS connection, and then check that the correct local address was propagated to the backend as the remote address? I'm confused on how that makes sure the TLS call failed

Reply:
Hehe, I guess it was too late when I worked on it and my mind was somewhere else - I've copied test from the previous one and started changing it for port=0 scenario, but didn't actually finish. 😅 Thanks for catching it.
Fixed in  https://github.com/gravitational/teleport/pull/31622/commits/e5f2e68a6f58c8acaee3afc7eceed7e7b9998bdb

Reply:
NP, thanks for fixing!